crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+                    .unwrap_or_default()
+                    .status_reason
+                    .unwrap_or_default();
+                Some(types::ErrorResponse {
+                    code: reason_info.reason.code,
+                    message: reason_info.reason.reject_reason.unwrap_or_default(),
+                    reason: None,
+                    status_code,
+                })
+            } else {
+                None
+            };
+            let payment_response_data = types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::NoResponseId,
+                redirection_data: None,
+                mandate_reference: None,
+                connector_metadata: None,
+            };
+            Ok((status, error, payment_response_data))
+        }
+        _ => Err(errors::ConnectorError::ResponseDeserializationFailed.into()),
+    }
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+pub struct TrustpayAuthUpdateRequest {
+    pub grant_type: String,
+}
+
+impl TryFrom<&types::RefreshTokenRouterData> for TrustpayAuthUpdateRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &types::RefreshTokenRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            grant_type: "client_credentials".to_string(),
+        })
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub struct ResultInfo {
+    pub result_code: i64,
+    pub additional_info: Option<String>,
+    pub correlation_id: Option<String>,
+}
+
+#[derive(Default, Debug, Clone, Deserialize, PartialEq)]
+pub struct TrustpayAuthUpdateResponse {
+    pub access_token: Option<String>,
+    pub token_type: Option<String>,
+    pub expires_in: Option<i64>,
+    #[serde(rename = "ResultInfo")]
+    pub result_info: ResultInfo,
+}
+
+#[derive(Default, Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub struct TrustpayAccessTokenErrorResponse {
+    pub result_info: ResultInfo,
+}
+
+impl<F, T> TryFrom<types::ResponseRouterData<F, TrustpayAuthUpdateResponse, T, types::AccessToken>>
+    for types::RouterData<F, T, types::AccessToken>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, TrustpayAuthUpdateResponse, T, types::AccessToken>,
+    ) -> Result<Self, Self::Error> {
+        match (item.response.access_token, item.response.expires_in) {
+            (Some(access_token), Some(expires_in)) => Ok(Self {
+                response: Ok(types::AccessToken {
+                    token: access_token,
+                    expires: expires_in,
+                }),
+                ..item.data
+            }),
+            _ => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: item.response.result_info.result_code.to_string(),
+                    message: item
```

Comment:
do not use default empty value here, use consts::NO_ERROR_MESSAGE

Reply:
Done